docs/examples/good_practices/wandb_setup/main.py

```diff
@@ -0,0 +1,204 @@
+"""Single-GPU training example."""
+import logging
+import os
+from pathlib import Path
+
+import rich.logging
+import torch
+from torch import Tensor, nn
+from torch.nn import functional as F
+from torch.utils.data import DataLoader, random_split
+from torchvision import transforms
+from torchvision.datasets import CIFAR10
+from torchvision.models import resnet18
+from tqdm import tqdm
+import wandb
+
+
+def main():
+    training_epochs = 10
+    learning_rate = 5e-4
+    weight_decay = 1e-4
+    batch_size = 128
+
+    # Check that the GPU is available
+    assert torch.cuda.is_available() and torch.cuda.device_count() > 0
+    device = torch.device("cuda", 0)
+
+    # Setup logging (optional, but much better than using print statements)
+    logging.basicConfig(
+        level=logging.INFO,
+        handlers=[rich.logging.RichHandler(markup=True)],  # Very pretty, uses the `rich` package.
+    )
+
+    logger = logging.getLogger(__name__)
+
+    # To resume experiments with Wandb, we need to have code that can properly
+    # handle checkpointing (see other minimalist example about "checkpointing").
+    # We have to manage the `id` of the experiment that we are running so that
+    # it is unique and Wandb knows what previous run came before this one
+    # (i.e. what is being resumed). This is handled in the same way that saving
+    # model parameters is handled.
+    # This specific example here does not do that.
+
+    # Setup Wandb
+    run = wandb.init(
+        # Set the project where this run will be logged
+        project="awesome-wandb-example",
+        name=os.environ.get("SLURM_JOB_ID"),
+        resume="allow",  # See https://docs.wandb.ai/guides/runs/resuming
+        # Track hyperparameters and run metadata
+        config={
+            "learning_rate": learning_rate,
+            "epochs": training_epochs,
+            "weight_decay": weight_decay,
+            "batch_size": batch_size,
+        },
+    )
+
+    # Create a model and move it to the GPU.
+    model = resnet18(num_classes=10)
+    model.to(device=device)
+
+    optimizer = torch.optim.AdamW(model.parameters(), lr=learning_rate, weight_decay=weight_decay)
+
+    # Setup CIFAR10
+    num_workers = get_num_workers()
+    dataset_path = Path(os.environ.get("SLURM_TMPDIR", ".")) / "data"
+    train_dataset, valid_dataset, test_dataset = make_datasets(str(dataset_path))
+    train_dataloader = DataLoader(
+        train_dataset,
+        batch_size=batch_size,
+        num_workers=num_workers,
+        shuffle=True,
+    )
+    valid_dataloader = DataLoader(
+        valid_dataset,
+        batch_size=batch_size,
+        num_workers=num_workers,
+        shuffle=False,
+    )
+    test_dataloader = DataLoader(  # NOTE: Not used in this example.
+        test_dataset,
+        batch_size=batch_size,
+        num_workers=num_workers,
+        shuffle=False,
+    )
+
+    # Check out the "checkpointing and preemption" example for more info!
```

Comment:
Maybe we could delete this line since we are the checkpointing example?
```suggestion
```

Reply:
> since we are the checkpointing example?

What do you mean?

Reply:
Sorry, _since we are in* checkpointing example_, why would we recommend again to check the "checkpointing and preemption" example?

Reply:
I still don't understand, this is the wandb example, what do you mean by "this is in the checkpointing example"?